runtime/src/main/java/org/corfudb/runtime/Backup.java

```diff
@@ -156,33 +185,42 @@ private void backup() throws IOException {
      * If the log is trimmed at timestamp, the backupTable will fail.
      * If the table has no data to be backed up, it will create a file with empty contents.
      *
-     * @param fileName   - the name of the backup file
+     * @param filePath   - the path of the backup file
      * @param uuid       - the uuid of the table which is being backed up
      * @throws IOException
      */
-    private void backupTable(String fileName, UUID uuid) throws IOException {
-        try (FileOutputStream fileOutput = new FileOutputStream(fileName)) {
+    private void backupTable(Path filePath, UUID uuid) throws IOException {
+        int numberOfEntries;
+        long startTime = System.currentTimeMillis();
+
+        try (FileOutputStream fileOutput = new FileOutputStream(filePath.toString())) {
             StreamOptions options = StreamOptions.builder()
                     .ignoreTrimmed(false)
                     .cacheEntries(false)
                     .build();
             Stream<OpaqueEntry> stream = (new OpaqueStream(runtime, runtime.getStreamsView().get(uuid, options))).streamUpTo(timestamp);
 
-            writeTableToFile(fileOutput, stream, uuid);
+            numberOfEntries = writeTableToFile(fileOutput, stream, uuid);
         } catch (IOException e) {
-            log.error("failed to back up table {} to file {}", uuid, fileName);
+            log.error("failed to back up table {} to file {}", uuid, filePath.toString());
             throw e;
         } catch (TrimmedException e) {
             log.error("failed to back up tables as log was trimmed after back up starts.");
             throw e;
         }
 
-        log.info("{} table is backed up and stored to temp file {}", uuid, fileName);
+        long elapsedTime = System.currentTimeMillis() - startTime;
+        backupTimeDistributionSummary.ifPresent(summary -> summary.record(elapsedTime));
```

Comment:
Better wrap the code block from line 194 to 212 in runnable and just record the runnable with timer.